block-producer/Cargo.toml

```diff
@@ -42,7 +43,7 @@ tracing-subscriber = { workspace = true }
 
 [dev-dependencies]
 figment = { version = "0.10", features = ["toml", "env", "test"] }
-miden-mock = { package = "miden-mock", git = "https://github.com/0xPolygonMiden/miden-base", branch = "main" }
+miden-mock = { workspace = true }
```

Comment:
reverted

Reply:
This is causing issues, because the types no longer match:

```
Error:   --> block-producer/src/block_builder/prover/tests.rs:48:13
   |
48 |             block_header,
   |             ^^^^^^^^^^^^ expected `miden_objects::BlockHeader`, found `BlockHeader`
   |
   = note: `BlockHeader` and `miden_objects::BlockHeader` have similar names, but are actually distinct types
note: `BlockHeader` is defined in crate `miden_objects`
  --> /home/runner/.cargo/git/checkouts/miden-base-c72873dca0cede8b/6e315e5/objects/src/block/header.rs:26:1
   |
26 | pub struct BlockHeader {
   | ^^^^^^^^^^^^^^^^^^^^^^
note: `miden_objects::BlockHeader` is defined in crate `miden_objects`
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/miden-objects-0.1.0/src/block/header.rs:26:1
   |
26 | pub struct BlockHeader {
   | ^^^^^^^^^^^^^^^^^^^^^^
   = note: perhaps two different versions of crate `miden_objects` are being used?
```

Reply:
ah - i see. i guess we have two options:

- publish `miden-mock` to crates.io (probably under a more specific name).
- get rid of the `miden-mock` dependency. what do we use it for here?

Reply:
>  what do we use it for here?

Tests: https://github.com/0xPolygonMiden/miden-node/blob/main/block-producer/src/block_builder/prover/tests.rs#L3

Reply:
If this is the only place, then I'm leaning toward removing the `miden-mock` dependency. There are a couple of ways to do it:

1. Duplicate the `mock_block_header` function in miden node (basically just copy it over from miden base).
2. Add a mock constructor to the `BlockHeader` struct (i.e., `BlockHeader::mock()`) and expose it under `testing` flag.

The second approach seems cleaner - but not sure how much changes this results in on the miden base side.

Reply:
So the strategy moving on is to not use the `miden-mock`, but to move its code to the other `miden-base` crates and put them behind a feature flag?